### ISSUE TITLE
Keep loading when adding my nextcloud account

### ISSUE THREAD

**Issue by MattRGX** [stefan-niedermann/nextcloud-deck#1547]
**Describe the bug**

Hello,

I'm trying to install Deck on my android device (Pixel 7). I already have the nextcloud files app (working perfectly) as well as other apps (notes, memories, money buster, cook, ...), all working perfectly.

When starting Deck, the app detect well my nextcloud files account. Then i click to add and it start loading my boards.

I only have 3 differents boards (they are not big at all).

But it keeps loading and get stuck always when loading the second board.  and then it get stuck in an infinite loading.

Thanks for your help.

**Versions**

 - Nextcloud: 27.1.4
 - Nextcloud Deck: 1.11.1
 - Nextcloud Android: 3.26.0
 - Nextcloud Android Deck: 1.23.4

**Smartphone (please complete the following information):**
- Device: Pixel 7
- Android-Version: Android 14
- App-Store:
  - [ ] Google Play Store
  - [ ] Google Play Store (Beta channel)
  - [x] F-Droid
  - [ ] Huawei AppGallery

**Comment by MattRGX** [comment]
_**can you define "infinite"? How long did you wait?**_

I would say between 10 to 15 minutes with screen ON on the app (caffein mode)

**_How's your internet connection? Do you have a strong Wi-Fi or a shitty mobile connection?_**

Yeah, i've tried both 5G or strong WiFI (i live in Paris).

**_Can you roughly tell us the total amount of cards you have? 10? 100? 1000? 10000?_**

I would say a total of around 50 to 70 cards (cumulated over my 3 boards)

**_Did you keep the app in foreground while importing or switch to other apps / homescreen etc?_**

stay on the screen of the app

**_Did you clear the storage of the Deck Android app as described in our FAQ between your attempts?_**

uninstall and reinstall the app multiple times instead. Is it the same result as clear the storage ?

**Comment by stefan-niedermann** [comment]
Thank you very much for the fast answers. That's indeed interesting, it definitely takes too long.

> uninstall and reinstall the app multiple times instead. Is it the same result as clear the storage ?

No, it is not (thank you Google).

Apparently there's an error somewhere which does not show up (as it should). Let's try to fence it. Please proceed as described in the FAQ:

> Try to create a new test account at your Nextcloud instance and connect to it. If there are no errors, share the boards of your actual account one by one with your test account to find the defect one.

**Comment by MattRGX** [comment]
ok i will do that, by the way, i forgot to mention that it always get stuck when loading the boards number 3. It is the one that contains most of the cards. I would say 40 to 50 cards in this one. 
So it might be that board that is related to my problem ?

**Comment by molivier-fcms** [comment]
Hi. I encounter the exact same problem, and I have identified a specific board that is blocking the initial loading. Can I help somehow as well?

**Versions**

- Nextcloud: 27.1.3
- Nextcloud Deck:  1.11.2
- Nextcloud Android: 3.26.0
- Nextcloud Android Deck: 1.23.4

**Smartphone** 

- Device: Pixel 6a
- Android-Version: Android 14

**App-Store**

- F-Droid

**Comment by MattRGX** [comment]
@molivier-fcms did you solved your issue ?
Because on my side, i have identified as well the board that cause the infinite loading. 
I've tried to export the board (as csv) from my nextcloud web interface, but then there is no import to create a board from CSV.
So i'm kinda stuck. 
The weird thing overall is that the board which cause trouble in Android is totally accessible and editable on the NC web interface.

**Comment by MattRGX** [comment]
@molivier-fcms you mean logcat with Adb while the infinite screen is ON ?

**Comment by MattRGX** [comment]
If it can help, i give you here logcat of my phone while infinite loading. 
[logcat.txt](https://github.com/stefan-niedermann/nextcloud-deck/files/13550972/logcat.txt)

I don't have the time to wait for a solution because i really need Deck on my phone. So i will copy the board card by card and completely create a new one, no choice...

Anyway thanks for your help and do not hesitate to ask if you want me to make some test :)

**Comment by molivier-fcms** [comment]
> I don't have the time to wait for a solution because i really need Deck on my phone. So i will copy the board card by card and completely create a new one, no choice...

It seems a deleted board remains until nextcloud cron run. You should run it before re-initializing deck, or you could stuck on the old invisible board again.

**Comment by MattRGX** [comment]
> > I don't have the time to wait for a solution because i really need Deck on my phone. So i will copy the board card by card and completely create a new one, no choice...
> 
> It seems a deleted board remains until nextcloud cron run. You should perform it before re-initializing deck, or the app could stuck while loading the old "invisible not yet deleted" board again ([#452](https://github.com/nextcloud/deck/issues/452))

you mean perform the nextcloud cron task ? If i go to my admin panel on the web interface, it says that the CRON have just run seconds ago, so i guess that it runs correctly.

i still have weird behaviour. i create a complete new board identical as the one that cause problem. i deleted the old one and clean everything. Now when trying to load deck on my phone it says that it is loading 5 boards (and get stucked at the 3rd one) but in fact i only have 3 boards.

I've clean all data / cache of Deck app then reinstall it but it keeps want to load 5 boards.

How can i clear what the android app is viewing ? because apparently the app got some wrong information from my server.

**Comment by MattRGX** [comment]
@molivier-fcms i thought about it too. But didn't make it yet. You conviced me to not touch any postgres table 😁

**Comment by molivier-fcms** [comment]
@MattRGX : Cards are related to stacks, and stacks are related to boards. Here is an exemple of a complete board removal: [#1163](https://github.com/nextcloud/deck/issues/1163#issuecomment-1515940941).

I tried it on my side and it worked : Deck app downloads the number of active boards, no more. Be sure to backup your database before, just in case.

**Comment by MattRGX** [comment]
> @MattRGX : Cards are related to stacks, and stacks are related to boards. I finally found out this example of a complete board removal: [#1163](https://github.com/nextcloud/deck/issues/1163#issuecomment-1515940941).
> 
> I tried it on my side and it worked : Deck app downloads the number of active boards, no more. Be sure to backup your database before, just in case.

so i've cleaned everything in database. All the boards / stacks / cards in my db are actually the ones that i use on my web interface. No old boards or cards left. 

But i still cant connect the deck app. it is stuck while importing the third board (out of 3). 

I'm kinda desperate ...

**Comment by molivier-fcms** [comment]
So you have recreated all cards from 3rd board, then removed the 3rd board.

If the issue persist, maybe there is a corruption in a card content (title/description)? Do you have special characters, such as "&" or something else? Could it be possible there is an invisible or a special symbol that hangs the import process?

Another track : try to re-initialize the board several times (with whole app cleaning in `Settings > Applications > Nextcloud Deck > Storage and Cache > Clear storage`). If you succeed once, clone the faulty board with all cards, then remove the old one in database, and try a new initialization. It could eventually ignore the faulty "pattern" (I do a very hazardous guess here).

**Comment by MattRGX** [comment]
yep all the 3 boards i have now are new, by new i mean they have been created after i've noticed the problem with the app. The old boards a completely deleted in DB. All the data in DB is only related to those 3 boards.

the weird thing is sometimes, after multiples app cleaning (as you mentioned) it get stuck loading board number 2, sometimes it get stuck loading board number 3 (never board number 1).  

You really think it can be at a card level ?

**Comment by molivier-fcms** [comment]
I would be surprised if deck had a structure problem (I mean inconsistency in database). Moreover since you recreated everything.
As nobody else complains about a similar issue, I wonder if the problem could be content related. Give a look at freshest cards you created eventually, their title, descriptions, comments.

**Comment by MattRGX** [comment]
100% of my cards only have a title, i never put any comment nor content inside (i'm the only user with basic needs of deck).
i have almost 100 cards (cumulated over the 3 boards). i don't know what can i check then

**Comment by MattRGX** [comment]
so i finally decided to export my 3 boards as csv, then delete all boards, uninstall Deck from my nextcloud instance, then reinstall it, and recreate my 3 boards one by one, and now everything work fine.
We will never know what was the cause of my problem. 
Thanks anyway for the help :)

Issue can be closed.

**Comment by molivier-fcms** [comment]
As long as it works. :+1: 
You don't have a button "Close issue" below comment box?
